mteb/abstasks/AbsTask.py

```diff
@@ -32,6 +34,31 @@ def dataset_transform(self):
         """
         pass
 
+    def stratified_subsampling(
+        self,
+        splits: Union[str, list[str]] = ["test"],
+        label: str = "label",
+        n_samples: int = max_n_samples,
+    ):
+        """Subsamples the dataset with stratification by the supplied label.
+        The following kwargs must be provided
+        for stratified_subsampling to run:
+        - splits: Union[str, list[str]], the splits of the dataset.
+        - label: str, the label with which the stratified sampling is based on.
+        - n_samples: Optional[int], number of samples to subsample. Default is max_n_samples.
+        """
+        if isinstance(splits, str):
+            splits = [splits]
+
+        ## Can only do this if the label column is of ClassLabel.
+        if not isinstance(self.dataset[splits[0]].features[label], datasets.ClassLabel):
+            self.dataset = self.dataset.class_encode_column(label)
+
+        for split in splits:
+            self.dataset[split] = self.dataset[split].train_test_split(
+                test_size=n_samples, seed=self.seed, stratify_by_column=label
+            )["test"]  ## only take the specified test split.
```

Comment:
I would instead create the dataset and return it and convert to method to a @staticmethod. This avoids state which can often lead to bugs.